guides/source/active_record_basics.md

```diff
@@ -220,134 +265,196 @@ The application will have to use the [`id_value`][] alias attribute to access th
+For example, given a `Book` model with attributes of `title` and `author`,
 the `create` method call will create and save a new record into the database:
 
 ```ruby
-user = User.create(name: "David", occupation: "Code Artist")
+book = Book.create(title: "The Lord of the Rings", author: "J.R.R. Tolkien")
 ```
 
-Using the `new` method, an object can be instantiated without being saved:
+While the `new` method will instantiate an object without saving it to the database:
 
 ```ruby
-user = User.new
-user.name = "David"
-user.occupation = "Code Artist"
-```
+book = Book.new
+book.title = "The Hobbit"
+book.author = "J.R.R. Tolkien"
+
+# The above `book` is not yet saved to the database
 
-A call to `user.save` will commit the record to the database.
+book.save
+
+# Now the `book` record is committed to the database.
+```
 
 Finally, if a block is provided, both `create` and `new` will yield the new
 object to that block for initialization, while only `create` will persist
 the resulting object to the database:
 
 ```ruby
-user = User.new do |u|
-  u.name = "David"
-  u.occupation = "Code Artist"
+book = Book.new do |b|
+  b.title = "Metaprogramming Ruby 2"
+  b.author = "Paolo Perrotta"
 end
+
+book.save
+```
+
+The resulting SQL statement from both `book.save` and `Book.create` look something like this:
+
+```sql
+/* Note that `created_at` and `updated_at` are automatically set */
+
+INSERT INTO "books" ("title", "author", "created_at", "updated_at") VALUES (?, ?, ?, ?) RETURNING "id"  [["title", "Metaprogramming Ruby 2"], ["author", "Paolo Perrotta"], ["created_at", "2024-02-22 20:01:18.469952"], ["updated_at", "2024-02-22 20:01:18.469952"]]
 ```
 
 ### Read
 
-Active Record provides a rich API for accessing data within a database. Below
-are a few examples of different data access methods provided by Active Record.
+Active Record provides a rich API for accessing data within a database. You can query a single record or multiple records, filter them by any attribute, order them, group them, select specific fields, and do anything you can do with SQL.
 
 ```ruby
-# return a collection with all users
-users = User.all
+# Return a collection with all books
+books = Book.all
+
+# Return a single book
+first_book = Book.first
+last_book = Book.last
+random_book = Book.take
 ```
 
-```ruby
-# return the first user
-user = User.first
+The above results in the following SQL:
+
+```sql
+# Book.all
+SELECT "books".* FROM "books"
+
+# Book.first
+SELECT "books".* FROM "books" ORDER BY "books"."id" ASC LIMIT ?  [["LIMIT", 1]]
+
+# Book.last
+SELECT "books".* FROM "books" ORDER BY "books"."id" DESC LIMIT ?  [["LIMIT", 1]]
+
+# Book.take
+SELECT "books".* FROM "books" LIMIT ?  [["LIMIT", 1]]
 ```
 
+We can also find specific books with `find_by` and `where`:
+
 ```ruby
-# return the first user named David
-david = User.find_by(name: 'David')
+# Return the first book by the given author
+book = Book.find_by(author: "J.R.R. Tolkien")
```

Comment:
This is very subjective, but teaching Rails I've found that people often confuse `find_by` and `where` when starting out. Since in this example of the `Book` model the `author` string is logically likely to be repeated and the `title` string is less likely to be repeated, I feel it would be better to illustrate the example by searching for the first book by title rather than by author. It is a case more similar to real life, since when you use `find_by` you generally search for more unique things as opposed to `where`.

Also, I don't know if it will be so redundant, but I think it would be good to specifically emphasize that `find_by` returns the first record and `where` returns an array of records in the line above, before the code comments.

Well, this same case is repeated a few times below, personally I would also suggest changing it to `title` instead of `author`.

What do you think?

Reply:
I like the idea of changing to `title` for `find_by`. It's a good point about adding the distinction between what's returned - a single record vs. an array.